gengokit/middlewares/middlewares_test.go

```diff
@@ -0,0 +1,236 @@
+	if wantFormatted != serviceFormatted {
+		t.Fatalf("Serivce middleware different than expected:\n\n%s", diff)
+	}
+}
+
+func TestNewEndpoitnMiddleware(t *testing.T) {
+	var wantEndpoints = `
+		package middlewares
+
+		import (
+			"github.com/go-kit/kit/endpoint"
+		)
+
+		func InjectEndpointMiddlewares(in endpoint.Endpoint) endpoint.Endpoint {
+			return in
+		}
+	`
+
+	_, data, err := generalService()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	middleware := New()
+	endpoints, err := middleware.Render(EndpointsPath, data)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	endpointsBytes, err := ioutil.ReadAll(endpoints)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wantFormatted, endpointFormatted, diff := thelper.DiffGoCode(wantEndpoints, string(endpointsBytes))
+	if wantFormatted != endpointFormatted {
+		t.Fatalf("Endpoints middleware different than expected:\n\n%s", diff)
+	}
+}
+
+func TestRenderPrevService(t *testing.T) {
+	var wantService = `
+		package middlewares
+
+		import (
+		pb "github.com/TuneLab/go-truss/gengokit/general-service"
+		)
+
+		func InjectServiceMiddlewares(in pb.ProtoServiceServer) pb.ProtoServiceServer {
+			return in
+		}
+
+		func FooBar() error {
+			return nil
+		}
+	`
+	_, data, err := generalService()
+	if err != nil {
+		t.Fail()
+	}
+
+	middleware := New()
+
+	middleware.LoadService(strings.NewReader(wantService))
+
+	service, err := middleware.Render(ServicePath, data)
+	if err != nil {
+		t.Fail()
+	}
+
+	serviceBytes, err := ioutil.ReadAll(service)
+	if err != nil {
+		t.Fail()
+	}
+
+	wantFormatted, serviceFormatted, diff := thelper.DiffGoCode(wantService, string(serviceBytes))
+	if wantFormatted != serviceFormatted {
+		t.Fatalf("Sevice middleware modified unexpectedly:\n\n%s", diff)
+	}
+}
+
+func TestRenderPrevEndpoints(t *testing.T) {
+	var wantEndpoints = `
+		package middlewares
+
+		import (
+			"github.com/go-kit/kit/endpoint"
+		)
+
+		func InjectEndpointMiddlewares(in endpoint.Endpoint) endpoint.Endpoint {
+			return in
+		}
+
+		func BarFoo(err error) bool {
+			if err != nil {
+				return true
+			}
+			return false
+		}
+	`
+
+	_, data, err := generalService()
+	if err != nil {
+		t.Fail()
+	}
+
+	middleware := New()
+
+	middleware.LoadEndpoints(strings.NewReader(wantEndpoints))
+
+	endpoints, err := middleware.Render(EndpointsPath, data)
+	if err != nil {
+		t.Fail()
+	}
+
+	endpointsBytes, err := ioutil.ReadAll(endpoints)
+	if err != nil {
+		t.Fail()
+	}
+
+	wantFormatted, endpointFormatted, diff := thelper.DiffGoCode(wantEndpoints, string(endpointsBytes))
+	if wantFormatted != endpointFormatted {
+		t.Fatalf("Endpoints middleware modified unexpectedly:\n\n%s", diff)
+	}
+}
+
+func TestRenderUnknownFile(t *testing.T) {
+	_, data, err := generalService()
+	if err != nil {
+		t.Fail()
+	}
+
+	middleware := New()
+
+	_, err = middleware.Render("not/valid/file.go", data)
+	if err == nil {
+		t.Fatalf("Tried to render unknown file")
```

Comment:
This doesn't mean much to a non-author. Add "this should have produced an error, but didn't" or something like that.